index.bs

```diff
@@ -485,6 +501,159 @@ spec:csp3; type:grammar; text:base64-value
   failed resource with a different one.
 
   <!-- ####################################################################### -->
+  
+  ## Integrity-Policy ## {#integrity-policy-section}
+  The `Integrity-Policy` and `Integrity-Policy-Report-Only` HTTP headers enable a document to
+  enforce a policy regarding the integrity metadata requirements on all the subresources it
+  loads of certain <a for=request>destinations</a>.
+
+  The headers' value is a <a>Dictionary</a> [[RFC9651]], with every member-value being an
+  <a>inner list</a> of <a>tokens</a>.
+  
+  A <dfn>source</dfn> is a string. The only possible value for it is "`inline`".
+
+  A <dfn>destination</dfn> is a <a for=request>destination</a>. The only possible value for it is "`script`".
+
+  An <dfn>integrity policy struct</dfn>, is a <a>struct</a> that contains the following:
+
+    * <dfn>sources</dfn>, a list of <a>source</a>s, initially empty.
+    * <dfn>blocked destinations</dfn>, a list of <a>destination</a>s, initially empty.
+    * <dfn>endpoints</dfn>, a <a for=/>list</a> of strings, initially empty.
+
+  When <dfn>processing an integrity policy</dfn>, with |headers| and |header name|, do the following:
```

Comment:
Since this is a new algorithm, also define the types of the arguments. And use camelcase for variable names. Otherwise it's hard to distinguish a single variable from multiple.

Reply:
Added!